File: 面试题10-I-斐波那契数列.py
class Solution:
    def fib(self, n: int) -> int:
        num1 = 1
        num2 = 0

        for i in range(n - 1):
            num1, num2 = (num1 + num2) % 1000000007, num1
        
        return num1 if n != 0 else 0

File: 面试题10-I-斐波那契数列.cpp
class Solution {
public:
    int fib(int n) {
        int num1 = 1, num2 = 0, num;

        for(int i = 0; i < n - 1; i++){
            num = num1;
            num1 = (num1 + num2) % 1000000007;
            num2 = num;
        }

        return n == 0 ? 0 : num1;
    }
};
